pennylane/shadows/classical_shadow.py

```diff
@@ -0,0 +1,181 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Classical Shadows baseclass with processing functions"""
+
+from collections.abc import Iterable
+import pennylane.numpy as np
+import pennylane as qml
+import warnings
+
+from collections.abc import Iterable
+
+
+class ClassicalShadow:
+    """TODO: docstring"""
+
+    def __init__(self, bitstrings, recipes):
+        self.bitstrings = bitstrings
+        self.recipes = recipes
+
+        assert len(bitstrings) == len(recipes)
+        self.snapshots = len(bitstrings)
+
+        self.unitaries = [
+            qml.matrix(qml.Hadamard(0)),
+            (qml.matrix(qml.Hadamard(0)) @ qml.matrix(qml.PhaseShift(np.pi / 2, wires=0))).conj().T,
+            qml.matrix(qml.Identity(0)),
+        ]
+
+    def local_snapshots(self, wires=None, snapshots=None):
```

Comment:
is there a way in python to have local snapshots be computed upon the first instance where they are necessary? and from then on use them as instance attributes?

Reply:
@Qottmann do you mean like a [property](https://realpython.com/python-property/)?

Reply:
yes, exactly, thanks! :)